NewSpring/Rock
# Updated page meta data lava increase performance

## DESCRIPTION

This PR adds a new lava template that handles the generation of metadata tags for pages from a persisted dataset. The lava that currently handles this is located in the "Page Meta Tags" block that is in every page on the site in production. This block will be updated upon deploy of this merge with an include to the new lava template: `{% include '~~/assets/lava/data/page-meta-data.lava' %}`

### Environment Configuration
1. Replace the contents of the `Page Meta Tags` block on any page of the external site in your environment with the new lava include above. Updating it on one page will update it on every page since it is a site-level block.
2. Configure the new persisted dataset in your environment as shown below:
![image](https://user-images.githubusercontent.com/2465823/78708710-8f46a300-78e0-11ea-807c-f762cbc45095.png)
3. Manually refresh the dataset once it has been created.

### Notes
Basically, if you read the lava template that is referenced in the page metadata block - this is what is supposed to happen.
1. If a page has a Content Component Block on it that uses the `Hero` template, reference allow that block to handle the adding of metadata tags.
2. If a page has its "Meta Data From Content Channel Item" attribute (under page settings > page attribute values) set to "Yes" it should allow the lava template that is used on the page to handle the adding of meta data tags (sermon detail layout, editorial layout, etc).
3. If the above two cases are both false, then the lava template adds meta tags to the page using the follow priority of values:
- Page Meta Title, if that doesn't have a value, Page Title
- Page Meta Description, if that doesn't have a value, Page Description
- Page Meta Image

### Testing
To test that this is working, you'll want to:
1. Visit a page with a Hero content component block. The metadata tags on this page (in the source code) should have the values from the hero block itself.
2. Visit a page that has meta data set from the content channel item (an article, sermon, story, etc). The metadata tags on this page (in the source code) should have the values from the content channel item itself.
3. Any other static page, should follow the priority listed in point #3 of the notes section above. You can update values in the various fields to verify the correct priority is being followed.

## TODO


## REVIEW


> The purpose of PR Review is to _improve the quality of the software._
